src/main/java/subway/domain/SubwayLine.java

```diff
@@ -13,58 +23,94 @@ public class SubwayLine {
     @Column(unique = true, nullable = false)
     private String name;
 
+    @Column(nullable = false)
     private String color;
 
-    private Long upStationId;
+    private Long distance = 0L;
 
-    private Long downStationId;
+    @ManyToOne
+    @JoinColumn(name = "up_station_id", nullable = false)
+    private Station upStation;
 
-    private Long distance;
+    @ManyToOne
+    @JoinColumn(name = "down_station_id", nullable = false)
+    private Station downStation;
 
-    @OneToMany(mappedBy = "subwayLine")
-    private List<Station> stations = new ArrayList<>();
+    @OneToMany(mappedBy = "subwayLine", cascade = CascadeType.ALL, orphanRemoval = true)
+    private List<Section> sections = new ArrayList<>();
 
-    public SubwayLine(String name, String color, Long upStationId, Long downStationId, Long distance) {
+    private SubwayLine(String name, String color) {
         this.name = name;
         this.color = color;
-        this.upStationId = upStationId;
-        this.downStationId = downStationId;
-        this.distance = distance;
     }
 
-    protected SubwayLine() {
+    public static SubwayLine of(String name, String color, Section section) {
+        var subwayLine = new SubwayLine(name, color);
+        subwayLine.addFirstSection(section);
+        return subwayLine;
     }
 
-    public void updateBasicInfo(String name, String color) {
-        this.name = name;
-        this.color = color;
+    private void addFirstSection(Section section) {
+        this.distance = section.getDistance();
+        this.upStation = section.getUpStation();
+        this.downStation = section.getDownStation();
+        this.sections.add(section);
+        section.assignSubwayLine(this);
     }
 
-    public Long getId() {
-        return id;
+    public void addSection(Section section) {
+        if (!canAddSection(section)) {
+            throw new UnsupportedOperationException();
+        }
+        this.distance = this.distance + section.getDistance();
+        this.downStation = section.getDownStation();
+        this.sections.add(section);
+        section.assignOrder(this.sections.size());
+        section.assignSubwayLine(this);
     }
 
-    public String getName() {
-        return name;
+
+    private boolean canAddSection(Section section) {
+        var sectionUpStationId = section.getUpStation().getId();
+        if (!Objects.equals(sectionUpStationId, this.downStation.getId())) {
+            return false;
+        }
+        var sectionDownStationId = section.getDownStation().getId();
+        return !this.hasStation(sectionDownStationId);
     }
 
-    public String getColor() {
-        return color;
+
+    public void updateBasicInfo(String name, String color) {
+        this.name = name;
+        this.color = color;
     }
 
-    public Long getUpStationId() {
-        return upStationId;
+    public void removeStation(Long stationId) {
```

Comment:
Line 입장에서 구간이 삭제되는 걸로 보는게 더 알맞을 것 같네용
```suggestion
    public void removeSection(Long stationId) {
```

Reply:
오 넵! 이게 더 자연스럽네요👍